src/SixLabors.Fonts/TextLayout.cs

```diff
@@ -1153,58 +1153,60 @@ VerticalOrientationType.Rotate or
         int maxLineBreakIndex = lineBreaks.Count - 1;
         LineBreak lastLineBreak = lineBreaks[lineBreakIndex];
         LineBreak currentLineBreak = lineBreaks[lineBreakIndex];
+        float lineAdvance = 0;
 
-        lineAdvance = 0;
         for (int i = 0; i < textLine.Count; i++)
         {
             int max = textLine.Count - 1;
             TextLine.GlyphLayoutData glyph = textLine[i];
             codePointIndex = glyph.CodePointIndex;
             int graphemeCodePointIndex = glyph.GraphemeCodePointIndex;
-            float glyphAdvance = glyph.ScaledAdvance;
-            lineAdvance += glyphAdvance;
 
             if (graphemeCodePointIndex == 0 && textLine.Count > 0)
             {
+                lineAdvance += glyph.ScaledAdvance;
+
                 if (codePointIndex == currentLineBreak.PositionWrap && currentLineBreak.Required)
                 {
                     // Mandatory line break at index.
                     TextLine remaining = textLine.SplitAt(i);
                     textLines.Add(textLine.Finalize(options));
                     textLine = remaining;
-                    i = 0;
+                    i = -1;
```

Comment:
This took me a few hours to figure out! Setting `i = 0` was incorrect as the loop was incrementing the value before we read the collection again.